src/coreComponents/common/TypeDispatch.hpp

```diff
@@ -254,30 +300,38 @@ bool dispatchViaTable( TypeList< Ts... > const types,
 
 /**
  * @brief Dispatch a generic worker function @p lambda based on runtime type.
- * @tparam Ts list of types
- * @tparam LAMBDA type of user-provided function or lambda, must have one auto argument
- * @param types list of types as an object (for deduction)
- * @param type type index of the runtime type
- * @param errorIfTypeNotFound flag indicating whether dispatch should issue an error when type not matched
- * @param lambda user-provided callable, will be called with a single prvalue of type indicated by @p type
+ *
+ * @tparam LIST type of the list of types
+ * @tparam LAMBDA  type of user-provided function or lambda, must have one auto argument
+ * @tparam Ts types of the objects to be dispatched.
+ * @param combinations list of types
+ * @param lambda lambda user-provided callable
+ * @param objects objects to be dispatched
  * @return @p true iff type has been dispatch
  *
- * @todo Do we want @p errorIfTypeNotFound parameter? Options:
+ * todo Do we want errorIfTypeNotFound parameter? Options:
  *       - make it a template parameter (caller always knows whether or not it wants hard errors)
  *       - make the caller process return value and raise error if needed (and however they want)
  */
-template< typename ... Ts, typename LAMBDA >
-bool dispatch( TypeList< Ts... > const types,
-               std::type_index const type,
-               bool const errorIfTypeNotFound,
-               LAMBDA && lambda )
+template< typename LIST, typename LAMBDA, typename ... Ts >
+bool dispatch( LIST const combinations,
+               LAMBDA && lambda,
+               Ts && ... objects )
 {
-  bool const success = internal::dispatchViaTable( types, type, std::forward< LAMBDA >( lambda ) );
-  if( !success && errorIfTypeNotFound )
+  bool const success = internal::dispatchViaTable( combinations,
+                                                   std::forward< LAMBDA >( lambda ),
+//                                                   std::make_tuple( std::type_index( internal::typeIdWrapper(objects))... ) );
```

Comment:
```suggestion
```

Reply:
How is this now working without using `internal::typeIdWrapper(objects)`?

Reply:
The unittest uses the base types, so I would expect that to pass with this, but yeah I'm surprised we're not erroring out when applying this to `WrapperBase` in the other locations. We don't have any mechanism to transparently unwrap the wrapped type, so I would expect we're `typeid()`-ing `WrapperBase` in all the non unit-test locations in this PR.

Reply:
I was keeping the failure to work on the error message